packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -80,64 +113,69 @@ class AgentRole(Dialogue.Role):
 
     @staticmethod
     def role_from_first_message(message: Message) -> Dialogue.Role:
-        """Infer the role of the agent from an incoming/outgoing first message
+        """
+        Infer the role of the agent from an incoming or outgoing first message
 
         :param message: an incoming/outgoing first message
-        :return: The role of the agent
+        :return: the agent's role
         """
         fipa_message = cast(FipaMessage, message)
-        return role_from_first_message(fipa_message)
+        if fipa_message.is_incoming:
+            return FipaDialogue.AgentRole.SELLER
+        else:
+            return FipaDialogue.AgentRole.BUYER
+        # return role_from_first_message(fipa_message)
```

Comment:
I had to put the code here for now due to circular import issue in custom_types.py